### PR TITLE
feat(crvd-3379): support for custom files

### PR DESCRIPTION
**Pull request by matanlb**

### CODE REVIEW

**Comment by nemni8** [comment]
I don't think we should change our current build flow yet. 
IMO we need to develop a parallel flow that will use the new wixData d.ts definitions without affecting the current solution. 
maybe even build 2 versions of corvid types. and then in order to get the new wix data types, we will just need to import the "new" bundle of corvid-types

File: customTypes/wix-data.d.ts [hunk 1]
```diff
@@ -0,0 +1,603 @@
+declare module "wix-data" {
```
Comment (nemni8) [inline customTypes/wix-data.d.ts:1]: too long for me. how do you know that those definitions work? we need to understand how we can test it..

File: scripts/generate-types.js [hunk 2]
```diff
@@ -23,35 +27,70 @@ const cloneDocsRepo = () => {
...
+  spawnSync("docworks", dockworksParams, {
+    stdio: "inherit"
+  });
 
   await $wGenerator(
     localDocsRepoPath,
     path.join(TYPES_PAGES_PATH, DECLARATION_FULL_FILE_NAME)
   );
+
+  if (!WITH_CUSTOM) return;
```
Comment (nemni8) [inline scripts/generate-types.js:78]: why not asking are there customFiles?

**Comment by matanlb** [comment]
@nemni8 quite a change you made me do 😅
but the more I've worked on it the more I saw it was necessary

I've tried to keep the custom-types mass as localised as I could but I had not choice but to split the index file to create a separate bundle without duplicating the code.

Testing wise we CAN tests, we can just import the build artefact like the normal types and run the same suite for it too. Only issue is that without converting the entire `wix-data` declaration file the tests file. (also there is an issue with `wix-router` having a small usage of `wix_data.WixDataQuery` that we'll need to solve) but I think we'll tackle this when we're back on this wix-data-autocomplete project. Until then we know that when we get to experiment stage we can have the tests run for these on every build once we decide to pull the trigger.

I know we're taking a pause from this but I'd really like to leave a clean env before we move on, so please try and re-review this :)

**Pull request closed without merging**